uwg_schema/model.py

```diff
@@ -236,25 +223,43 @@ class UWG(NoExtraBaseModel):
         'pedestrians, and street cooking.'
     )
 
-    bld: conlist(conlist(float, min_items=3, max_items=3), min_items=1) = Field(
+    bld: conlist(
+        conlist(
+            Union[float, str], min_items=3, max_items=3),
+        min_items=1) = Field(
         ...,
-        description='Matrix of numbers representing fraction of urban building stock. '
-        'This property consists of a 16 x 3 matrix referencing the fraction of the '
-        'urban building stock from 16 building types and 3 built eras representing, in '
-        'combination with 16 climate zones, 768 building archetypes generated from the '
-        'Commercial Building Energy Consumption Survey. Each column represent a '
-        'pre-1980s, post-1980s, or new construction era, and rows represent building '
-        'types. Custom build types can be added by adding new rows. The sum of the '
-        'fractional values in the bld matrix must sum to one.'
+        description='List of building types, eras, and fractions of urban building '
+        'stock. This consists of a list of tuples, each containing a string for the '
+        'building type, a string for the the built era, and a number between 0 and '
+        '1, inclusive, defining built stock fraction, i.e ("LargeOffice", "New", 0.4). '
+        'The fractions should sum to one. 768 predefined models are built referencing '
+        '16 building types for 3 built eras and 16 climate zones according to models '
+        'from the Department of Energy (DOE). Custom building types can also be '
+        'referenced in this property.'
     )
 
     @validator('bld')
     def check_bld(cls, value):
-        """Ensure bld matrix dimensions."""
-        for i in range(len(value)):
-            assert len(value[i]) == 3, 'The bld property must be a 16 (or greater) ' \
-                'x 3 matrix. Got {} columns for the row {}.'.format(
-                    len(value[i]), i)
+        """Ensure bld tuples have correct order of types."""
+        total_frac = 0.0
+        for bld_row in value:
+            bldtype, builtera, frac = bld_row[0], bld_row[1], bld_row[2]
+            assert isinstance(bldtype, str), 'The first item in the ' \
+                'bld tuple must be text defining the reference building ' \
+                'type. Got: {}.'.format(bldtype)
+            assert isinstance(builtera, str) and builtera.lower() in REF_BUILTERA_SET, \
+                'The second item in the bld tuple must be text defining the built ' \
+                'era as one of {}. Got: {}.'.format(
+                    REF_BUILTERA, builtera.lower())
```

Comment:
It's probably worth stating in the docstring for the `bld` matrix that there are only 3 acceptable options for builtera.  However, bldtype can either be one of the 16 standard types OR it can be anything that matches a bldtype in the Model's detailed BemDefs.

Reply:
Sure. My docstrings in the UWG.bld essentially says this. I was aiming to keep the schema docstrings shorter, but I can incorporate an abbreviated version here. 

```
"""Get or set list of building types, eras, and fractions of urban building stock.

        This property consists of a list of tuples, each containing a string for the the
        built era, and a number between 0 and 1, inclusive, defining built stock
        fraction, i.e ('LargeOffice', 'New', 0.4). The fractions should sum to one.

        768 predefined models are built referencing 16 building types for 3 built eras
        and 16 climate zones according to models from the Department of Energy (DOE).
        Choose from the following text identifiers to reference a DOE building type:

        * 'fullservicerestaurant'
        * 'hospital'
        * 'largehotel'
        * 'largeoffice'
        * 'medoffice'
        * 'midriseapartment'
        * 'outpatient'
        * 'primaryschool'
        * 'quickservicerestaurant'
        * 'secondaryschool'
        * 'smallhotel'
        * 'smalloffice'
        * 'standaloneretail'
        * 'stripmall'
        * 'supermarket'
        * 'warehouse'

        Choose from the following built eras:

        * 'pre80'
        * 'pst80'
        * 'new'

        Custom building types can also be referenced in this property. For example, a
        built stock consisting of 40% post-1980's large office, 30% new midrise
        apartment, and 30% of a pre=1980s custom building type (defined by the user)
        is referenced as follows:

        .. code-block:: python

            bld = [('largeoffice', 'pst80', 0.4),
                   ('midriseapartment', 'new', 0.3),
                   ('custombuilding', 'pre80', 0.3)]
        """
```